indexing-service/src/main/java/org/apache/druid/indexing/common/actions/SegmentTransactionalInsertAction.java

```diff
@@ -63,13 +62,16 @@
   private final DataSourceMetadata endMetadata;
   @Nullable
   private final String dataSource;
+  @Nullable
+  private final Set<DataSegment> segmentsToBeDropped;
```

Comment:
Can you add some javadocs to this class that explain the difference between `segmentsToBeOverwritten` and `segmentsToBeDropped`?

Reply:
Done